unity/Assets/Scripts/Util/DataStructures/BST/IBST.cs

```diff
@@ -64,5 +83,7 @@ public interface IBST<T> where T : IComparable<T>, IEquatable<T>
         /// Clears the tree of all nodes.
         /// </summary>
         void Clear();
+
+        IEnumerable<T> Iterator();
```

Comment:
Maybe make the Interface inherit form IEnumerable. Since the interface requires the implementation of an iterator it would be logical to inherit form IEnumerable

Reply:
Good idea, done. The `Iterator()` method was only added for easier debugging, but it's still better to specify it properly.